Use array size directly when checking multiscale arrays to prevent overflow

kephale opened the pull request:
# Fixes/Closes

Closes #5758

# Description

This PR changes the way array sizes are compared when guessing if data is multiscale.

The problem was that `np.prod(a.shape)` was being used which leads to an overflow (see Notes in [numpy.ndarray.size](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.size.html)). That has been changed to be `a.size` instead (ty @aganders3).

Additionally, during testing there was an issue with what seemed to be an out of memory error with the Python 3.8 Windows min req job (ty @jaimergp). As a result the test now also uses a Dask delayed array to avoid the large memory allocation (ty @aganders3).

## Type of change
<!-- Please delete options that are not relevant. -->
- [x] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?
<!-- Please describe the tests that you ran to verify your changes. -->
- [x] example: the test suite for my feature covers cases x, y, and z
- [x] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

kephale committed:
Use object dtype when computing size of arrays to prevent overflow

kephale committed:
Add test case for guess_multiscale size overflow

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

alisterburt reviewed (approved):
Nice! Looks good to me 🚀

kephale committed:
Apply @aganders3 fix

Fix was shared here: https://github.com/napari/napari/pull/5759#issuecomment-1526127464

kephale committed:
Fix typo
